components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/util/Utils.java

```diff
@@ -227,16 +226,18 @@ public static String getOrganizationId() {
      */
     public static String getAuthenticatedUsername() {
 
-        String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
-        if (username == null) {
+        String userResidentOrganizationId = PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .getUserResidentOrganizationId();
+        /* When user accessing a different organization, the user resident organization is populated in the carbon
+            context. That value can be used to find the username from the user ID. */
+        if (StringUtils.isNotEmpty(userResidentOrganizationId)) {
             try {
-                username = organizationUserResidentResolverService.resolveUserFromResidentOrganization(null,
-                        getUserId(), getOrganizationId()).map(User::getUsername).orElse(null);
-            } catch (OrganizationManagementException e) {
+                return getUserStoreManager(userResidentOrganizationId).getUser(getUserId(), null).getUsername();
```

Comment:
Isn't this a possible NPE is the user not found

Reply:
The user is tried to resolve based on the information in the carbon context. But there can be situations where different user-ids set (in fedarated user). Hence handling null check would be safe.
Fixed by - https://github.com/wso2/identity-organization-management-core/pull/108/commits/ea8a1ae9f6f03f588046a34c2d29af2724c3c178